tracer/src/Datadog.Trace/Configuration/SettingsManager.cs

```diff
@@ -0,0 +1,220 @@
+            // it can mean that _everything_ about the exporter settings changes. To minimize the work to do, and
+            // to simplify comparisons, we try to read the agent url from the manual setting. If it's missing, not
+            // set, or unchanged, there's no need to update the exporter settings.
+            // We only technically need to do this today if _manual_ config changes, not if remote config changes,
+            // but for simplicity we don't distinguish currently.
+            var exporterTelemetry = new ConfigurationTelemetry();
+            var newRawExporterSettings = ExporterSettings.Raw.CreateUpdatedFromManualConfig(
+                currentExporter.RawSettings,
+                manualSource,
+                exporterTelemetry,
+                manualSource.UseDefaultSources);
+
+            var isSameMutableSettings = currentMutable.Equals(newMutableSettings);
+            var isSameExporterSettings = currentExporter.RawSettings.Equals(newRawExporterSettings);
+
+            if (isSameMutableSettings && isSameExporterSettings)
+            {
+                Log.Debug("No changes detected in the new configuration");
+                // Even though there were no "real" changes, there may be _effective_ changes in telemetry that
+                // need to be recorded (e.g. the customer set the value in code, but it was already set via
+                // env vars). We _should_ record exporter settings too, but that introduces a bunch of complexity
+                // which we'll resolve later anyway, so just have that gap for now (it's very niche).
+                // If there are changes, they're recorded automatically in ConfigureInternal
+                telemetry.CopyTo(centralTelemetry);
+                return null;
+            }
+
+            Log.Information("Notifying consumers of new settings");
+            var updatedMutableSettings = isSameMutableSettings ? null : newMutableSettings;
+            var updatedExporterSettings = isSameExporterSettings ? null : new ExporterSettings(newRawExporterSettings, exporterTelemetry);
+
+            return new SettingChanges(updatedMutableSettings, updatedExporterSettings, currentMutable, currentExporter);
+        }
+
+        private void NotifySubscribers(SettingChanges settings)
+        {
+            // Strictly, for safety, we only need to lock in the subscribers list access. However,
+            // there's nothing to prevent NotifySubscribers being called concurrently,
+            // which could result in weird out-of-order notifications for customers. So for simplicity
+            // we just lock the whole method to ensure serialized updates.
+
+            lock (_subscribers)
+            {
+                var subscribers = _subscribers;
+                Volatile.Write(ref _latest, settings);
+
+                foreach (var subscriber in subscribers)
```

Comment:
What's the use of local variable `subscribers` instead of iterating `_subscribers` directly?

Reply:
I think this is just left over from when I wasn't doing a big lock in Subscribe() 🤔 I think we can just remove it